### DISCUSSION TITLE
Cannot query the list of cameras though the motion web interface

### DISCUSSION THREAD

**Discussion by pascal-fb-martin** [Motion-Project/motion#1829] [Q&A] [answered]
I am trying to integrate motion into my own web environment. I did not set a motion username or password. I can access the motion's web UI from a browser and stream the cameras, but all my "manual" requests (using firefox, targeting the motion control port 8080) for config information fail.

I searched through the motion code, and I found that URLs are decoded using the format /camid/cmd1/cmd2... I did find a supported URL with cmd1 = "config" and cmd2 = "list", but no matter what I tried for camid returns "Bad request".

I also found a reference to request "cameras.json", either as camid or cmd1, but that returns "Bad request" as well.

What exact HTTP request should I use?

(It would be nice to add a MD file that gives an overview of the Motion's web API, BTW.)

**Comment by Mr-Dave** [comment]
Are you referring to Motion or Motionplus?  (Motion does not provide any json pages and this discussion is in the Motion repro)

Edit:
Oops.  My bad.  It has been too long since I've looked at Motion.  It does have a json page.

Additional Reference for Motion:
[Webcontrol](https://motion-project.github.io/motion_config.html#OptDetail_Webcontrol)
[Stream](https://motion-project.github.io/motion_config.html#OptDetail_Stream)

**Reply by pascal-fb-martin** [reply]
Thanks for the edit. BTW, the Motion versions are 4.3.2 and 4.5.1 (I split my cameras between two RPIs. Upgrading the second one was a chore--cannot upgrade Raspbian in-situ--so I delayed the upgrade for one).

I checked: what I tried was {IP}:{port}/{camid}/config/list. I got a bit confused with what is used as an ID. I have a camera that is defined as:

```
camera_id 11
camera_name watch11
```

so I tried both "/11/config/list" and "/watch11/config/list". Each returns HTTP status 200, but Content-Length is 0.

I also realize now that the /config/list does not return the _list_ of cameras: I also tried /cameras.json (which should trigger a call to webu_status_list and then webu_status_write_list), but this returns HTTP status 200 with content "Bad Request .." (??).

Thanks.

**Reply by Mr-Dave** [reply]
I'd guess that the [webcontrol_parms](https://motion-project.github.io/motion_config.html#webcontrol_parms) has not be set.  Also make sure to consider that the cameras.json is on the stream port not the webcontrol port.

**Comment by pascal-fb-martin** [comment]
Motion, at https://github.com/Motion-Project/motion.git.

I retrieved the reference to cameras.json in function
webu_answer_strm_type(). I also found the /config/list URI referenced in
many places in webu.c, webu_text.c and webu_html.c

I also looked at MotionEye, and it seems to make use of the /config/list
URI.

Thanks.
Pascal.

**Reply by Mr-Dave** [reply]
Note my edit from my original response above:

> Edit:
Oops. My bad. It has been too long since I've looked at Motion. It does have a json page.
Additional Reference for Motion:
[Webcontrol](https://motion-project.github.io/motion_config.html#OptDetail_Webcontrol)
[Stream](https://motion-project.github.io/motion_config.html#OptDetail_Stream)


The documentation pages provide the various URLs/APIs

**Comment by Mr-Dave** [comment]
One Additional note so you are not surprised.

Although all the parameters may be listed, many of them really can't be effectively changed via the web control.  Hence, this is reason that Motioneye rewrites the configuration files and shuts down/restarts the Motion application.

If you need something that does process all the configuration parameters, I'd suggest Motionplus.

**Comment by pascal-fb-martin** [comment, marked as answer]
I noticed the webcontrol_parms item, which was left to 0. I set it to 2, as mentioned in the configuration documentation. Not sure what level I really need, but _/config/list now works_. /cameras.json worked with the 8081 port (stream). I got confused by this last detail..

Sorry for bothering you.
